src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,901 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports Microsoft.VisualBasic
+Imports Microsoft.VisualBasic.FileIO
+Imports System
+Imports System.Collections
+Imports System.Collections.Generic
+Imports System.Collections.ObjectModel
+Imports System.ComponentModel
+Imports System.Linq
+Imports System.Runtime.InteropServices
+Imports Xunit
+' Do not Imports System.IO
+
+
+Namespace Microsoft.VisualBasic.Tests
+    Public NotInheritable Class FileIOTests
+        ReadOnly DestData() As Char = {"x"c, "X"c, "y"c}
+
+        ReadOnly SourceData() As Char = {"a"c, "A"c, "b"c}
+
+        Sub New()
+        End Sub
+
+        Private Function CreateTestFile(TestBase As FileIOTestBase, ByVal TestData() As Char, Optional TestFileNameWithPath As String = "") As String
+            If TestFileNameWithPath.Length = 0 Then
+                TestFileNameWithPath = TestBase.GetTestFilePath()
+            Else
+                Assert.True(IO.Path.IsPathRooted(TestFileNameWithPath))
+                Assert.True(TestFileNameWithPath.StartsWith(TestBase.TestDirectory))
+            End If
+            ' Write and copy file
+            Using sourceStream As New IO.StreamWriter(IO.File.Create(TestFileNameWithPath))
+                sourceStream.Write(TestData, 0, TestData.Length)
+            End Using
+
+            Return TestFileNameWithPath
+        End Function
+
+        <Fact>
+        Public Sub CombinePathTest_BadBaseDirectory_RelativePath()
+            Assert.Throws(Of ArgumentNullException)(Function() FileSystem.CombinePath("", "Test2"))
+            Assert.Equal(FileSystem.CombinePath("C:\", ""), "C:\")
```

Comment:
why this assert is in this test ?

Reply:
This test is broken, completely wrong on anything but Windows.. I have replaced it. Thanks. There needs to be a test for adding a null or empty string to root. directory because it is a special case in the code